.github/workflows/auto-publish.yml

```diff
@@ -0,0 +1,25 @@
+name: Build, and publish spec to GitHub Pages and /TR/
+
+on:
+  workflow_dispatch: {}
+  pull_request: {}
+  push:
+    branches: [main]
+
+jobs:
+  main:
+    name: Build, Validate and Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
I think you'll start getting warnings/failures very soon for this - we should probably be using `actions/checkout@v4`. I've had other GHActions fail on this relatively recently.

Reply:
updated.
I'm not sure why,, it is said that v2 depends on older node.js (v12?) which is already disappeared from [ubuntu-latest](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md) (or some earlier images), but actually working on some other GHActions..
in any case, yes, switching newer on this front does not make any harm (we don't use any options) and should be welcome.